pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -101,6 +101,24 @@ type ScrapeConfigSpec struct {
 	// Authorization header to use on every scrape request.
 	// +optional
 	Authorization *v1.SafeAuthorization `json:"authorization,omitempty"`
+	// SampleLimit defines per-scrape limit on number of scraped samples that will be accepted.
+	// +optional
+	SampleLimit uint64 `json:"sampleLimit,omitempty"`
```

Comment:
We'll need those types to be pointer as per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required since `uint64` doesn't have a `nil` value.

Reply:
`AddLimitsToYAML` expects limit to be `uint64`, not `*uint64` and checks if it's not zero [here](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L208-L211).
So I assime it's better to remove `+optional` comment instead of modifying existing logic everywhere or dereference these pointers?

Reply:
I'd be in favor of updating `AddLimitsToYAML`()

Reply:
Since that would affect existing ServiceMonitor / PodMonitor / Probe behaviour as they use it as well, that goes above my confidence level in codebase changes :)

Some help would be appreciated in that case

Reply:
That's a valid concern @dmitriishaburov, thanks for sharing!

Maybe we could create a PR updating the old fields to pointer first @xiu @simonpasquier? Then a rebase on this one should make it easier

Reply:
Good idea, was thinking of a follow up at first but might as well do it beforehand :) Opened #5668 for that (still draft waiting for e2e to pass).